### PR TITLE
fix(protocol): say the state store is owned by the memory server and what to do without get_state

### PR DESCRIPTION
**Pull request by Fmarzochi**
## What Changed

`scripts/bootstrap-cognitive.js`: every variant of the memory protocol text (the Markdown block for Claude Code and Antigravity, the standalone Markdown file for OpenCode, Trae and CodeBuddy, the compact one-line form, and the Codex `persistent_instructions` string) now says:

- state is owned by the `egc-memory` server: encrypted at rest under the EGC home, one file per project and branch;
- the agent never reads or writes those files directly;
- when `get_state` is not among the agent's tools, the server is not registered for that tool: say so and point at `egc init` instead of keeping memory on the filesystem.

The sentence "State files live at `~/.egc/state/<project-slug>.md`: plain Markdown, one file per project" and its short forms are gone.

`tests/scripts/bootstrap-cognitive.test.js`: a regression case installs the block and the Codex line into a temp home and asserts the new wording is present and the old one is not.

## Why This Change

Refs #1395. In #1380 an Antigravity with no MCP servers registered answered a memory question by listing `~/.egc/state` and reading a `.md` with its file tool; the first line of that file is a Markdown heading, so the agent had written it itself in an earlier session, following the text we handed it. That file sits next to the encrypted ones, unencrypted, and every legacy reader picks it up.

## Scope note

The protocol version stays at 5 on purpose: no version bump was authorised. The new text therefore reaches fresh installs and any file that is upgraded for another reason; installs that already carry a v5 block keep their current text until the next protocol bump. #1395 stays open for that decision and for the optional doctor warning about plaintext state files.

## Testing Done

- [x] Manual testing completed
- [x] Automated tests pass locally (`node tests/run-all.js`)
- [x] Edge cases considered and tested
- [ ] If this PR touches a file shared across concurrent EGC processes (encryption key, state files, install-state, lockfiles under `~/.egc/`), a concurrent-access test was added or updated (not applicable: text only)

## Type of Change
- [x] `fix:` Bug fix

## Security & Quality Checklist
- [x] No secrets or API keys committed (ghp_, sk-, AKIA, xoxb, xoxp patterns checked)
- [x] JSON files validate cleanly
- [x] Shell scripts pass shellcheck (if applicable)
- [x] Pre-commit hooks pass locally (if configured)
- [x] No sensitive data exposed in logs or output
- [x] Follows conventional commits format

## Documentation
- [x] Updated relevant documentation (the protocol text is the documentation)
- [x] Added comments for complex logic
- [ ] README updated (not needed)

## Credit

Found through @Akisolu's follow-up checks in #1380.

---
## Summary by cubic
Fixes the memory protocol text so agents stop treating state files as plain Markdown at a filesystem path. The old wording pointed agents at `~/.egc/state/<slug>.md`; an agent without `get_state` registered would read and write that file unencrypted. The new wording in every variant says the store is owned by `egc-memory`, encrypted at rest, one file per project and branch, never read or written directly, and to say the server is not registered and point at `egc init` when `get_state` is missing.

- Adds a regression test asserting the new wording in the Claude Code block, Codex `persistent_instructions`, standalone Markdown, and Cursor rules.
- Protocol version stays at 5: fresh installs get the new text, existing installs keep their block until the next protocol bump.

<sup>Written for commit e1a9a5ddb0e7863e872ac25fe7b6f61e8b0374da. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/Fmarzochi/EGC/pull/1398?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: tests/scripts/bootstrap-cognitive.test.js [hunk 1]
```diff
@@ -123,6 +123,28 @@ async function runClaudeCodeAndGeminiCliTests() {
     }
   })) passed++; else failed++;
 
+  if (await test('protocol says the state store is owned by the server and what to do without get_state, in the block and Codex forms (#1395)', () => {
```
[the remark this answers is not in the document]
Comment (Fmarzochi) [reply tests/scripts/bootstrap-cognitive.test.js:126]: Taken in 6400ee96: the same test now installs into an OpenCode home and a Cursor settings.json and asserts the new wording, and the absence of the old one, in the standalone Markdown file and in cursor.rules as well.
